config/options.go

```diff
@@ -125,12 +126,43 @@ func defaultResolver(input map[string]interface{}) error {
 	return resolve(input)
 }
 
-func expand(s string, mapping func(string) string) string {
+func convertToType(input string) interface{} {
+	// Check if the input is a string with quotes
+	if strings.HasPrefix(input, "\"") && strings.HasSuffix(input, "\"") {
+		// Trim the quotes and return the string value
+		return strings.Trim(input, "\"")
+	}
+
+	// Try converting to bool
+	if input == "true" || input == "false" {
+		b, _ := strconv.ParseBool(input)
+		return b
+	}
+
+	// Try converting to float64
+	if strings.Contains(input, ".") {
+		if f, err := strconv.ParseFloat(input, 64); err == nil {
+			return f
+		}
+	}
+
+	// Try converting to int64
+	if i, err := strconv.ParseInt(input, 10, 64); err == nil {
+		return i
+	}
+
+	// Default to string if no other conversion succeeds
+	return input
+}
+func expand(s string, mapping func(string) string) interface{} {
```

Comment:
@shenqidebaozi How do you suggest to make it an optional feature?

Reply:
Yes, because before the feature is added, all configurations will be parsed into strings, and users may have code to handle the conversion of strings to other types. If we modify the default behavior, it may cause exceptions for users.

Reply:
I agree with your point to make it an optional behaviour. 

Making it optional
* Adding a option to enable this feature https://github.com/go-kratos/kratos/blob/main/config/options.go#L23 
wdyt?

Reply:
We can leave the Resolver method definition unchanged and only make modifications within the internal implementation

Reply:
```
func newDefaultResolver(o bool) func(map[string]interface{}) error {
	return func(input map[string]interface{}) error {
		// do something
	}
}
```

Reply:
You can refer to the code above

Reply:
NewDefaultResolver needs to be exported only then user will be able to make it configured like 
```
c := config.New(
        config.WithResolver(config.NewDefaultResolver(true)),
        config.WithSource(
            file.NewSource(flagconf),
        ),
    )
```

Reply:
We can export a configuration method instead of exporting the entire implementation

Reply:
@go-kratos/contributor please review

Reply:
@akoserwal We don't want to expose the underlying implementation, we just need to add a `WithResolveSpecific Type`

Reply:
please fix lint

Reply:
> @akoserwal We don't want to expose the underlying implementation, we just need to add a `WithResolveSpecificType`

or `ResolveActualTypes`

Reply:
The actual type resolver can be configured like:

```	c := config.New(
		config.WithResolveActualTypes(true),
		config.WithSource(
			file.NewSource(flagconf),
		),
	)
```

Reply:
@shenqidebaozi Please take a look again